libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -124,30 +128,24 @@ class NavigationView @JvmOverloads constructor(
 
         MapboxNavigationApp.attach(this)
 
-        val componentActivity = context.toComponentActivity()
-        attachCreated(
-            AnalyticsComponent(),
-            LocationPermissionComponent(componentActivity, navigationContext.store),
-            TripSessionComponent(lifecycle, navigationContext.store),
-            MapLayoutCoordinator(navigationContext, binding),
-            BackPressedComponent(
-                componentActivity.onBackPressedDispatcher,
-                navigationContext.store,
-                lifecycleOwner = this,
-            ),
-            ScalebarPlaceholderCoordinator(navigationContext, binding.scalebarLayout),
-            ManeuverCoordinator(navigationContext, binding.guidanceLayout),
-            InfoPanelCoordinator(
-                navigationContext,
-                binding.infoPanelLayout,
-                binding.guidelineBottom
-            ),
-            ActionButtonsCoordinator(navigationContext, binding.actionListLayout),
-            SpeedLimitCoordinator(navigationContext, binding.speedLimitLayout),
-            RoadNameCoordinator(navigationContext, binding.roadNameLayout),
-            LeftFrameCoordinator(navigationContext, binding.emptyLeftContainer),
-            RightFrameCoordinator(navigationContext, binding.emptyRightContainer)
-        )
+        val activity = context.toComponentActivity()
+        navigationContext.apply {
+            attachCreated(
+                analyticsComponent(),
+                locationPermissionComponent(activity),
+                tripSessionComponent(),
+                mapLayoutCoordinator(binding),
+                backPressedComponent(activity),
+                scalebarPlaceholderCoordinator(binding.scalebarLayout),
+                maneuverCoordinator(binding.guidanceLayout),
+                infoPanelCoordinator(binding.infoPanelLayout, binding.guidelineBottom),
+                actionButtonsCoordinator(binding.actionListLayout),
+                speedLimitCoordinator(binding.speedLimitLayout),
+                roadNameCoordinator(binding.roadNameLayout),
+                leftFrameCoordinator(binding.emptyLeftContainer),
+                rightFrameCoordinator(binding.emptyRightContainer)
+            )
+        }
```

Comment:
Unfortunately, I was unable to test the `MapboxNavigationApp.setup()` call.
For some reason, Mockk doesn't detect the matcher for `MapboxNavigationApp.setup(NavigationOptions)`:

```
mockkObject(MapboxNavigationApp)
every { MapboxNavigationApp.setup(ofType(NavigationOptions::class)) } returns MapboxNavigationApp
```

When ran, it returns an error:
```
Failed matching mocking signature for
SignedCall(retValue=, isRetValueMock=true, retType=class com.mapbox.navigation.core.lifecycle.MapboxNavigationAppDelegate, self=MapboxNavigationApp(object MapboxNavigationApp), method=getMapboxNavigationAppDelegate(), args=[], invocationStr=MapboxNavigationApp(object MapboxNavigationApp).getMapboxNavigationAppDelegate())
SignedCall(retValue=, isRetValueMock=true, retType=class com.mapbox.navigation.core.lifecycle.MapboxNavigationAppDelegate, self=, method=setup(NavigationOptionsProvider), args=[com.mapbox.navigation.core.lifecycle.MapboxNavigationApp$$Lambda$220/0x0000000800c3d840@5e2c3382], invocationStr=MapboxNavigationAppDelegate(temporary mock).setup(com.mapbox.navigation.core.lifecycle.MapboxNavigationApp$$Lambda$220/0x0000000800c3d840@5e2c3382))
left matchers: [ofType(NavigationOptions)]
io.mockk.MockKException: Failed matching mocking signature for
SignedCall(retValue=, isRetValueMock=true, retType=class com.mapbox.navigation.core.lifecycle.MapboxNavigationAppDelegate, self=MapboxNavigationApp(object MapboxNavigationApp), method=getMapboxNavigationAppDelegate(), args=[], invocationStr=MapboxNavigationApp(object MapboxNavigationApp).getMapboxNavigationAppDelegate())
SignedCall(retValue=, isRetValueMock=true, retType=class com.mapbox.navigation.core.lifecycle.MapboxNavigationAppDelegate, self=, method=setup(NavigationOptionsProvider), args=[com.mapbox.navigation.core.lifecycle.MapboxNavigationApp$$Lambda$220/0x0000000800c3d840@5e2c3382], invocationStr=MapboxNavigationAppDelegate(temporary mock).setup(com.mapbox.navigation.core.lifecycle.MapboxNavigationApp$$Lambda$220/0x0000000800c3d840@5e2c3382))
left matchers: [ofType(NavigationOptions)]
...
```
It appears that when we call `mockkObject(MapboxNavigationApp)`, Mockk generates a temporary mock for `MapboxNavigationAppDelegate` to which we don't have access.

Reply:
You should use `mockkStatic`, not `mockkObject`. Because the method is annotated as `@JvmStatic`.